--- a/util.go
+++ b/util.go
@@ -181,6 +181,14 @@
 	if httpErr, ok := errors.AsType[*googleapi.Error](err); ok {
 		statusCode = httpErr.Code
 		fmt.Fprintf(&msg, "%d %s: %s", httpErr.Code, http.StatusText(httpErr.Code), httpErr.Message)
+		if httpErr.Code == http.StatusForbidden {
+			for _, item := range httpErr.Errors {
+				switch item.Reason {
+				case "accessDenied", "billingNotEnabled", "blocked":
+					adbcErr.Code = adbc.StatusUnauthorized
+				}
+			}
+		}
 	} else if apiErr, ok := errors.AsType[*apierror.APIError](err); ok {
 		// Despite all the structure inside the error, there isn't a great way to
 		// extract or map it onto anything (e.g. there are two types of errors
